vocode/streaming/streaming_conversation.py

```diff
@@ -249,7 +252,20 @@ async def process(self, item: InterruptibleEvent[AgentResponse]):
                     ):
                         await self.conversation.filler_audio_worker.wait_for_filler_audio_to_finish()
 
-                self.conversation.logger.debug("Synthesizing speech for message")
+                self.conversation.logger.debug("Synthesizing speech for message: " + agent_response_message.message.text)
+
+                # Check if the message indicates that we should press a DTMF key.  If so, and if the conversation supports it, play that tone.
+                # This happens in addition to normal processing, which means we will play the tone and also speak the message.
+                self.conversation.logger.debug("Checking if message shows intent to press DTMF key: %s" % agent_response_message.message.text)
+                match = DTMF_INTENT_REGEX.match(agent_response_message.message.text.lower())
```

Comment:
what do you think about having the agent respond with a new `AgentResponseType` for pressing digits?

https://github.com/vocodedev/vocode-python/blob/f1a2056dc6eb67469cdb8f3716dba23230515e08/vocode/streaming/agent/base_agent.py#L36